pkg/ruler/ruler_ring.go

```diff
@@ -96,16 +71,9 @@ func (cfg *RingConfig) ToLifecyclerConfig(logger log.Logger) (ring.BasicLifecycl
 	}, nil
 }
 
-func (cfg *RingConfig) ToRingConfig() ring.Config {
-	rc := ring.Config{}
-	flagext.DefaultValues(&rc)
-
-	rc.KVStore = cfg.KVStore
-	rc.HeartbeatTimeout = cfg.HeartbeatTimeout
-	rc.SubringCacheDisabled = true
-
-	// Each rule group is loaded to *exactly* one ruler.
-	rc.ReplicationFactor = 1
-
-	return rc
+func (cfg *RingConfig) ringOptions() []util.RingOption {
```

Comment:
Why not just keeping `ToRingConfig()` as we had before, overriding the parent one.

[nit] I would adopt this pattern in the overrides-exporter ring config too.